packages/tabs/src/TabTitle/TabTitle.tsx

```diff
@@ -20,89 +23,78 @@ import {
+  ) => {
+    const baseFontSize: BaseFontSize = useUpdatedBaseFontSize();
+    const { Component } = useInferredPolymorphic(as, rest, 'button');
+    const { index, ref, id } = useDescendant(TabDescendantsContext);
+    const { tabPanelDescendants } = useTabPanelDescendantsContext();
 
-  const theme = darkMode ? Theme.Dark : Theme.Light;
-  const selected = index === selectedIndex;
+    const theme = darkMode ? Theme.Dark : Theme.Light;
+    const selected = index === selectedIndex;
 
-  const relatedTabPanel = useMemo(() => {
-    return tabPanelDescendants.find(
-      tabPanelDescendant => tabPanelDescendant.index === index,
-    );
-  }, [tabPanelDescendants, index]);
+    const relatedTabPanel = useMemo(() => {
+      return tabPanelDescendants.find(
+        tabPanelDescendant => tabPanelDescendant.index === index,
+      );
+    }, [tabPanelDescendants, index]);
 
-  const handleClick = useCallback(
-    (event: React.MouseEvent) => {
-      onClick(event, index);
-    },
-    [index, onClick],
-  );
+    const handleClick = useCallback(
+      (event: React.MouseEvent) => {
+        onClick(event, index);
+      },
+      [index, onClick],
+    );
 
-  const nodeText = getNodeTextContent(rest.name);
+    const nodeText = getNodeTextContent(rest.name);
 
-  const sharedTabProps = {
-    ...rest,
-    className: cx(
-      listTitleFontSize[baseFontSize],
-      listTitleStyles,
-      listTitleModeStyles[theme].base,
-      {
-        [listTitleModeStyles[theme].selected]: !disabled && selected,
-        [listTitleModeStyles[theme].hover]: !disabled && !selected,
-        [listTitleModeStyles[theme].disabled]: disabled,
-      },
-      listTitleModeStyles[theme].focus,
-      className,
-    ),
-    disabled,
-    id,
-    name: nodeText,
-    onClick: handleClick,
-    role: 'tab',
-    tabIndex: selected ? 0 : -1,
-    ['aria-controls']: relatedTabPanel?.id,
-    ['aria-selected']: !disabled && selected,
-    ['data-text']: nodeText,
-  } as const;
+    const componentProps = {
+      ...rest,
+      className: cx(
+        listTitleFontSize[baseFontSize],
+        listTitleStyles,
+        listTitleModeStyles[theme].base,
+        {
+          [listTitleModeStyles[theme].selected]: !disabled && selected,
+          [listTitleModeStyles[theme].hover]: !disabled && !selected,
+          [listTitleModeStyles[theme].disabled]: disabled,
+        },
+        listTitleModeStyles[theme].focus,
+        className,
+      ),
+      disabled,
+      id,
+      name: nodeText,
+      onClick: handleClick,
+      ref: fwdRef,
+      role: 'tab',
+      tabIndex: selected ? 0 : -1,
+      ['aria-controls']: relatedTabPanel?.id,
+      ['aria-selected']: !disabled && selected,
+      ['data-text']: nodeText,
+    } as const;
 
-  if (typeof rest.href === 'string') {
     return (
-      <Box
-        as="a"
-        ref={titleRef as RefObject<HTMLAnchorElement>}
-        {...sharedTabProps}
-      >
+      <Component {...componentProps}>
```

Comment:
technically i believe this now allows `Tab` to render "as" any other element, not just `a` or `button`. Should add this to the changeset (minor)